internal/configuration/validator/const.go

```diff
@@ -43,12 +59,19 @@ const (
 	testTLSCert       = "/tmp/cert.pem"
 	testTLSKey        = "/tmp/key.pem"
 
-	errAccessControlInvalidPolicyWithSubjects = "Policy [bypass] for rule #%d domain %s with subjects %s is invalid. It is " +
-		"not supported to configure both policy bypass and subjects. For more information see: " +
+	errAccessControlInvalidPolicyWithSubjects = "Policy [bypass] for rule #%d domain %s with subjects %s is invalid. " +
+		"It is not supported to configure both policy bypass and subjects. For more information see: " +
 		"https://www.authelia.com/docs/configuration/access-control.html#combining-subjects-and-the-bypass-policy"
 )
 
 var validLoggingLevels = []string{"trace", "debug", "info", "warn", "error"}
+var validScopes = []string{"openid", "email", "profile", "groups", "offline_access"}
+var validOIDCGrantTypes = []string{"implicit", "refresh_token", "authorization_code", "password", "client_credentials"}
+var validOIDCResponseTypes = []string{
+	"code", "code id_token", "id_token", "token id_token", "token", "token id_token code",
```

Comment:
what about all permutations of `token id_token code`?

Reply:
I removed this. I think we need to carefully check the spec before we implement this. Your comment actually triggered this thought. I believe there are only certain orders/combinations that are technically accepted by the spec, though it would be good to validate this one way or another.

Reply:
Ok great.